membership-levels/limit-members-level.php

```diff
@@ -0,0 +1,249 @@
+<?php
+/**
+ * Limit the number of members in a level.
+ * This is a very large snippet that provides a complete solution for limiting the number of members in a membership level
+ * -- Show the limit in various places
+ * -- Hide levels that reach the limit from the public
+ * -- Redirecting away from checkout when the limit is reached.
+ * -- Confirm there are spots available before checking out.
+ *
+ * Note: This does not affect pre-existing members that had a level before this code is implemented.
+ *
+ * title: Limit Members Per Membership Level
+ * layout: snippet
+ * collection: membership-levels
+ * category: limit-members
+ *
+ * You can add this recipe to your site by creating a custom plugin
+ * or using the Code Snippets plugin available for free in the WordPress repository.
+ * Read this companion article for step-by-step directions on either method.
+ * https://www.paidmembershipspro.com/create-a-plugin-for-pmpro-customizations/
+ */
+
+/**
+ * Save Level Meta for a member limit.
+ */
+function my_pmpro_limit_members_pmpro_save_membership_level( $save_id ) {
+	$pmpro_level_member_limit = intval( $_REQUEST['pmpro_level_member_limit'] );
+	update_pmpro_membership_level_meta( $save_id, 'pmpro_level_member_limit', $pmpro_level_member_limit );
+}
+add_action( 'pmpro_save_membership_level', 'my_pmpro_limit_members_pmpro_save_membership_level', 10, 1 );
+
+/**
+ * Add Level Meta for a member limit.
+ */
+function my_pmpro_limit_members_pmpro_membership_level_after_other_settings() {
+	$edit_level_id = $_REQUEST['edit'];
+	$pmpro_level_member_limit = get_pmpro_membership_level_meta( $edit_level_id, 'pmpro_level_member_limit', true );
+	?>
+	<table class="form-table">
+		<tbody>
+			<tr>
+				<th scope="row" valign="top"><label for="pmpro_level_member_limit"><?php esc_html_e( 'Maximum Number of Members', 'pmpro-customizations' ); ?></label></th>
+				<td>
+					<input id="pmpro_level_member_limit" name="pmpro_level_member_limit" type="number" value="<?php echo esc_attr( $pmpro_level_member_limit );?>" />
```

Comment:
we could leverage type number here and add "min" attribute to avoid negative numbers